src/types/unit_true.rs

```diff
@@ -0,0 +1,64 @@
+use serde::de::{self, Deserialize, Deserializer, Visitor};
+use serde::ser::{Serialize, Serializer};
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub struct True;
+
+impl<'de> Deserialize<'de> for True {
+    fn deserialize<D>(des: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>
+    {
+        des.deserialize_bool(TrueVisitor)
+    }
+}
+
+struct TrueVisitor;
+
+impl<'de> Visitor<'de> for TrueVisitor {
+    type Value = True;
+
+    fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+        formatter.write_str("bool, equal to `true`")
+    }
+
+    fn visit_bool<E>(self, value: bool) -> Result<Self::Value, E>
+    where
+        E: de::Error
+    {
+        match value {
+            true => Ok(True),
+            _ => Err(E::custom("expected `true`, found other"))
```

Comment:
You visit `bool` here, so you can write
```suggestion
            false => Err(E::custom("expected `true`, found `false`"))
```

Reply:
Ok